src/v/cluster/partition_manager.cc

```diff
@@ -162,10 +163,26 @@ ss::future<> partition_manager::stop_partitions() {
     co_await _gate.close();
     // prevent partitions from being accessed
     auto partitions = std::exchange(_ntp_table, {});
-    _raft_table.clear();
+
+    {
+        auto current = _raft_table.begin();
+        while (current != _raft_table.end()) {
+            current = _raft_table.erase(current, ++current);
```

Comment:
Is this safe?

`erase(current, ++current)` both modifies current and uses it in another argument which I believe is indeterminately sequenced (this changed in C++17 from being UB to "indeterminately sequenced" but that stronger guarantee isn't very useful here). That is you might end up with `(current + 1), (current + 1)` (empty range) or `(current), (current + 1)`  (what you want).

In any case why use this range erase overload  at all over simply `_raft_table.erase(current++)`?